core/bash_impl.py

```diff
@@ -300,6 +300,27 @@ def SparseArray_SetElement(sparse_val, index, s):
     return 0
 
 
+def SparseArray_UnsetElement(sparse_val, index):
+    # type: (value.SparseArray, mops.BigInt) -> int
+    """A non-zero return value represents an error code. If this
+    function returns 1, the specified index is out of range.
+
+    """
+
+    index, ok = _SparseArray_CanonicalizeIndex(sparse_val, index)
+    if not ok:
+        return 1  # error_code: out-of-range index
+    mylib.dict_erase(sparse_val.d, index)
+
+    # update max_index
+    if mops.Equal(index, sparse_val.max_index):
+        sparse_val.max_index = mops.MINUS_ONE
+        for index in sparse_val.d.keys():
```

Comment:
I think you can remove `keys()` here -- it's actually less efficient in Python 2, and in mycpp

i.e. `for index in sparse_val.d` will just iterate over the dictionary keys

while  `.keys()` makes a copy of it, and then iterates over it

Reply:
Oh, I haven't thought that `Dict` itself would iterate the keys. Thank you. Also, I've naively thought `.keys()` would return an iterator. I'll fix it later.

Reply:
I fixed it and rebased the PR on top of the latest master.